pkg/operator/apiserver/controller/auditpolicy/auditpolicy_controller.go

```diff
@@ -0,0 +1,136 @@
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/yaml"
+)
+
+type AuditPolicyController struct {
+	apiserverConfigLister                configv1listers.APIServerLister
+	kubeClient                           kubernetes.Interface
+	operatorClient                       v1helpers.OperatorClient
+	targetNamespace, targetConfigMapName string
+}
+
+// NewAuditPolicyController create a controller that watches the config.openshift.io/v1 APIServer object
+// and reconciles a ConfigMap in the target namespace with the audit.k8s.io/v1 policy.yaml file.
+func NewAuditPolicyController(
+	targetNamespace string,
+	targetConfigMapName string,
+	apiserverConfigLister configv1listers.APIServerLister,
+	operatorClient v1helpers.OperatorClient,
+	kubeClient kubernetes.Interface,
+	configInformers configinformers.SharedInformerFactory,
+	kubeInformersForTargetNamesace kubeinformers.SharedInformerFactory,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &AuditPolicyController{
+		operatorClient:        operatorClient,
+		apiserverConfigLister: apiserverConfigLister,
+		kubeClient:            kubeClient,
+		targetNamespace:       targetNamespace,
+		targetConfigMapName:   targetConfigMapName,
+	}
+
+	return factory.New().WithSync(c.sync).ResyncEvery(10*time.Second).WithInformers(
+		configInformers.Config().V1().APIServers().Informer(),
+		kubeInformersForTargetNamesace.Core().V1().ConfigMaps().Informer(),
+	).ToController("AuditPolicyController", eventRecorder.WithComponentSuffix("audit-policy-controller"))
+}
+
+func (c *AuditPolicyController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	operatorConfigSpec, _, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+
+	switch operatorConfigSpec.ManagementState {
+	case operatorsv1.Managed:
+	case operatorsv1.Unmanaged:
+		return nil
+	case operatorsv1.Removed:
+		errs := []error{}
+		if err != nil {
+			errs = append(errs, err)
+			return errors.NewAggregate(errs)
+		}
+		if err := c.kubeClient.CoreV1().ConfigMaps(c.targetNamespace).Delete(ctx, c.targetConfigMapName, metav1.DeleteOptions{}); err != nil {
+			errs = append(errs, err)
+		}
+		return errors.NewAggregate(errs)
+	default:
+		syncCtx.Recorder().Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorConfigSpec.ManagementState)
+		return nil
+	}
+
+	config, err := c.apiserverConfigLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	err = c.syncAuditPolicy(ctx, config.Spec.Audit, syncCtx.Recorder())
+
+	// update failing condition
+	cond := operatorv1.OperatorCondition{
+		Type:   "AuditPolicyDegraded",
+		Status: operatorv1.ConditionFalse,
+	}
+	if err != nil {
+		cond.Status = operatorv1.ConditionTrue
+		cond.Reason = "Error"
+		cond.Message = err.Error()
+	}
+	if _, _, updateError := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(cond)); updateError != nil {
+		if err == nil {
+			return updateError
+		}
```

Comment:
i think we want to log the `updateError` if `err != nil`

Reply:
we have this style 15 times in library-go. This is just copied boilerplate :)